fix(db): Translation to Ello Quest - Add Stitches Completion Event

LannyE opened the pull request:
In classic Wow after the Translation to Ello quest is completed an map-wide
event would commence.  This is the event where the Stitches creature
would make a run from across the area map from Raven Hill to Darkshire, eliminating the
Night Watchers/Players in its path.

Currently the area event does not exist and does not trigger.  

Fix includes data from https://github.com/TrinityCore/TrinityCore/issues/22369 and Azerothcore #68

<!-- First of all, THANK YOU for your contribution.
 Please fill this template unless your PR is very simple/straightforward.
 Do not forget to have a look at our Pull Request tutorial: http://www.azerothcore.org/wiki/Contribute#how-to-create-a-pull-request
-->

<!-- WRITE A RELEVANT TITLE -->

## Changes Proposed:
Creation of the Stitches Event which is comprised of:
- New dialogue for Town Crier, Lord Ello Ebonlocke, Althea Ebonlocke, Stitches, and Watcher Cutford (Blizzlike)
- Broadcasts made to players of the map on status of event (Blizzlike)
- Content for Night Watch Camp (Blizzlike)
- Content for the Rotting Orchard entrance (Blizzlike)
- Content for Darkshire defense (Blizzlike)
- Stiches run across the map (Blizzlike)
- Watcher Cutford  run across the map (Blizzlike)
- Different NPC reactions and resets upon Stitches death at different points on the map 
- Stopping the event (Town Crier officially stops event but event will also stop after 30 minutes)
- Repositioned Night Watchers into more Blizzlike locations and orientations.
- Watcher Petras now patrols like he should (Blizzlike)
- Removed level 65 generic Night Watchers (non-Blizzlike)
- Remove old Watcher Cutford GUID.  Now SAI manages Watcher Cutford.


## Issues Addressed:
- Fixes #68 
<!-- If the issue doesn't exist, describe it and how to reproduce it, please. If the issue already exists, just paste the link to the issue you close, like this: Closes https://github.com/azerothcore/azerothcore-wotlk/issues/967 -->


## Tests Performed:
-  Completed Quest 252: Translation to Ello
Watch for Watcher Cutford
![WoWScrnShot_020821_192557](https://user-images.githubusercontent.com/65646022/107420242-ea838c00-6ad5-11eb-9668-8255db130433.jpg)

He talks with Lord Ello and then runs out and talks with Althea.  He then begins his run all the way to the Night Watch camp as Stitches is spawned and begins his run to Darkshire. 

Watcher Corwin spawn only for event
![WoWScrnShot_020821_192828](https://user-images.githubusercontent.com/65646022/107420966-b2c91400-6ad6-11eb-9f66-89a70dcd6a9c.jpg)

Watcher Sarys spawn only for event
![WoWScrnShot_020821_191602](https://user-images.githubusercontent.com/65646022/107420160-d3dd3500-6ad5-11eb-85e6-e2fb4d268689.jpg)

Night Watch Camp (Watcher Blomberg and Watcher Hutchins event only watchers) gets stomped as Cutford arrives.  He does not engage but runs into camp (Blizzlike).   Watcher Dodds and Paige respawn after event.
![WoWScrnShot_020821_191126](https://user-images.githubusercontent.com/65646022/107421249-f4f25580-6ad6-11eb-90e6-b80f4088b71b.jpg)

Stitches move to the Rotting Orchard path entrance (Watchers Thayer, Gelwin, Selkin, Merant event only watchers) and stomps those four night watchers after they move into position
![WoWScrnShot_020821_191252](https://user-images.githubusercontent.com/65646022/107421359-1c492280-6ad7-11eb-90da-0556400f8f7d.jpg)

Night Watchers form in Darkshire and engage Stitches ending the event (Watchers respwn after event except for Jordan and Keefer who follow normal respawn rules.)
![WoWScrnShot_020821_191535](https://user-images.githubusercontent.com/65646022/107421571-5c100a00-6ad7-11eb-8bb1-bc84b92a501b.jpg)


<!-- Does it build without errors? Did you test in-game? What did you test? Did you do all these tests on Linux, Mac or Windows? Other tests performed -->

 - Merges without issue with AzerothCore rev. 71b40ed8bc75+ 2021-02-08 14:52:03 +0000 (master branch)
- Checked GUIDs in game_event, Smart_Scripts, Creature_Templete, Creature, Creature_Addon, Waypoints, and Creature_Texts  

## How to Test the Changes:
<!-- We need to confirm the changes first, so try to make the work easy for testers (who are not necessarily coders), please:
 - Which commands to use? Which NPC to teleport to?
 - Do we need to enable debug flags on Cmake?
 - Do we need to look at the console? etc...
 - Other steps
-->
- Go to Darkshire
- .quest remove 252
- .quest add 252
- Turn in quest to Lord Ello Ebonlocke
- Event should start with Watcher Cutford running in a few seconds later. 


## Known Issues and TODO List:
<!-- This is a TODO list with checkboxes to tick -->
- [ Night Watcher formations or makeup may not be 100% accurate]



## Target Branch(es):
- [x] Master


<!-- NOTE: You do not need to squash your commits, on merge we will squash them for you (when there are too many commits we merge them into one big commit for a cleaner and easy to read history). -->

<!-- NOTE2: If you intend to contribute more than once, you should really join us on our discord channel!
 The link is on our site http://azerothcore.org/ We set cosmetic ranks for our contributors and may give access to special resources/knowledge to them! -->


<!-- Do not remove the instructions below about testing, they will help users to test your PR -->
 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

BarbzYHOOL commented:
hey, thank you for this big PR!

Can you bash run this https://github.com/azerothcore/azerothcore-wotlk/blob/master/apps/git_tools/setup_git_commit_template.sh for next PRs? it gives a template for the commit names/desc

Voxstrasza commented:
```
/* Affected rows: 471  Found rows: 0  Warnings: 0  Duration for 130 queries: 0.031 sec. */
```

few issues with the SAI, I'll be posting a video once it finishes uploading

Voxstrasza commented:
Here's the video of my testing:

https://youtu.be/BA_QwAslp74

Watchers Thayer, Gelwin, Selkin, Merant did not spawn and Stitches is borderline unkillable once he reaches Darkshire.  Once he stomps all the watchers, he'll wonder around until more watchers show up.

LannyE commented:
Thank for the video.  That helps alot.  Vox is I could burden you again for more testing I would appreciate it.  

Some details: The script uses a game event to manage stitches.  It has a 30 minute (a guess, we can change it) duration then he will despawn and the event will reset.  So if he breaks through the defenders he will only stay another 15 - 20 minutes curb stomping guards/players.    The game event is started once Watcher Cutford finishes talking to Commander Althea.  At that point I need you to check a few things.  

First:  When stitches arrives in Darkshire I noticed two Watchers missing.  Watchers Corwin and Sarys.  They spawn with the game event.  They should show in their spawnpoints at that time.  I have pictures in this thread showing their locations in the Inn and at the Blacksmiths.  Let me know if they show up. 

Second: The four watchers at the The Rotten Orchard entrance spawn (invisible) down the path on the other side of the big tree.  Once stitches stomps the Night Watch camp they run to their last stand location at the Orchard entrance.  You saw one of them in your video at 5:13
![watchers](https://user-images.githubusercontent.com/65646022/107557757-3e09de80-6b97-11eb-9dc0-bf8bf7037565.JPG)

Yes, that looks like a forsaken guy in the video and it probably is.  The forsaken guy probably will not be there, but would you please position yourself there and wait for Stitches to get past the Night Watch Camp where it broadcasts "The Night Watch Camp was Overrun".  At that point four watchers should emerge and run to the Orchard entrance.  I suspect the forsake guy will appear, if he wasn't already there and no watchers.  Let me know what you see. 

As far as the running goes I can change that to a walk.  

If you see more forsaken/horde creatures in the place of the watchers it is either a bug in SAI that changes the model of the watcher and sends it to its home location.  I have been fighting that bug/feature since I started this a week ago and it is random and hard to pin down.   Nowhere in the SAI scripts am I telling the creature to change modelID so I am not sure what is going on.  I think we will pursue this if you see them again.  

Let me know.   Thanks

LannyE commented:
While I am thinking about it.  The issue where a Watcher will change ModelID to a Forsaken/Horde and then run back to a home location happened alot.  It would only happen to certain Watchers even though other Watchers would run the exact same SAI code and work fine.   Watcher Hutchins is one that gave me trouble.  This is the SQL for the Smart Scripts I used for him that was used for each other Watcher and worked fine for them:
``` SQL
-- Watcher Hutchins
UPDATE `creature_template` SET `AIName` = "SmartAI" WHERE `entry` = 1001;
DELETE FROM `smart_scripts` WHERE `entryorguid` = 1001 AND `source_type` = 0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`event_param5`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_param4`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES 
(1001,0,0,0,38,0,100,0,1,1,0,0,0,53,1,1001,0,0,0,2,1,0,0,0,0,0,0,0,0,'Watcher Hutchins - On Data Set - Start Waypoint (No Repeat)'),
(1001,0,1,2,38,0,100,0,1,2,0,0,0,101,1,0,0,0,0,0,8,0,0,0,0,-10912,-388,40.632,0.812,"Watcher Hutchins - On Data Set - Set Home POS"),
(1001,0,2,0,61,0,100,0,0,0,0,0,0,24,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Watcher Hutchins - On Data Set - Evade"),
(1001,0,3,0,69,0,100,0,85,0,0,0,0,70,2,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Watcher Hutchins - On Game Event \'Stitches\' End - Respawn"),
(1001,0,4,0,11,0,100,0,0,0,0,0,0,8,2,0,0,0,0,0,1,0,0,0,0,0,0,0,0,'Watcher Hutchins - On Spawn - React State Aggressive');
```
Basically if Stitches died before reaching the Night Watch Camp after Watcher Hutchins moved into formation he was to move back using evade.  It works fine for the other Watchers but not him.  

Here he is after he moved into formation once Stitches hit a specific Waypoint. 
![WoWScrnShot_020821_125459](https://user-images.githubusercontent.com/65646022/107569776-ec1c8500-6ba5-11eb-842f-1fbe1b2f00f3.jpg)

I then kill Stitches and everyone runs back except for Watch Hutchins who changes modelID then runs back to his home location due to him evading:
![WoWScrnShot_020821_091955](https://user-images.githubusercontent.com/65646022/107570014-31d94d80-6ba6-11eb-92f0-5611266a89ff.jpg)

You can see in the screenshot the NPC info says DisplayID 22380 (Native: 2388).  2388 is his correct modelID.  There is no reason to change the modelID.  

So being a rookie to Smart Scripts and the documentation is difficult and lacking crucial details, I don't know if this is a bug or how smart_scripts deals with errors.  It happened the most when a specific Watcher moved and then Stitches set data (stitches death) for that Watcher and the Watcher is told to move based on that data being set.  I guess I need to know if I should submit a bug report or change the smart scripts.

pklloveyou commented:
Waiting for test mark

pklloveyou commented:
Tested

Voxstrasza commented:
> 
> 
> While I am thinking about it. The issue where a Watcher will change ModelID to a Forsaken/Horde and then run back to a home location happened alot. It would only happen to certain Watchers even though other Watchers would run the exact same SAI code and work fine. Watcher Hutchins is one that gave me trouble. This is the SQL for the Smart Scripts I used for him that was used for each other Watcher and worked fine for them:
> 
> ```sql
> -- Watcher Hutchins
> UPDATE `creature_template` SET `AIName` = "SmartAI" WHERE `entry` = 1001;
> DELETE FROM `smart_scripts` WHERE `entryorguid` = 1001 AND `source_type` = 0;
> INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`event_param5`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_param4`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES 
> (1001,0,0,0,38,0,100,0,1,1,0,0,0,53,1,1001,0,0,0,2,1,0,0,0,0,0,0,0,0,'Watcher Hutchins - On Data Set - Start Waypoint (No Repeat)'),
> (1001,0,1,2,38,0,100,0,1,2,0,0,0,101,1,0,0,0,0,0,8,0,0,0,0,-10912,-388,40.632,0.812,"Watcher Hutchins - On Data Set - Set Home POS"),
> (1001,0,2,0,61,0,100,0,0,0,0,0,0,24,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Watcher Hutchins - On Data Set - Evade"),
> (1001,0,3,0,69,0,100,0,85,0,0,0,0,70,2,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Watcher Hutchins - On Game Event \'Stitches\' End - Respawn"),
> (1001,0,4,0,11,0,100,0,0,0,0,0,0,8,2,0,0,0,0,0,1,0,0,0,0,0,0,0,0,'Watcher Hutchins - On Spawn - React State Aggressive');
> ```
> 
> Basically if Stitches died before reaching the Night Watch Camp after Watcher Hutchins moved into formation he was to move back using evade. It works fine for the other Watchers but not him.
> 
> Here he is after he moved into formation once Stitches hit a specific Waypoint.
> ![WoWScrnShot_020821_125459](https://user-images.githubusercontent.com/65646022/107569776-ec1c8500-6ba5-11eb-842f-1fbe1b2f00f3.jpg)
> 
> I then kill Stitches and everyone runs back except for Watch Hutchins who changes modelID then runs back to his home location due to him evading:
> ![WoWScrnShot_020821_091955](https://user-images.githubusercontent.com/65646022/107570014-31d94d80-6ba6-11eb-92f0-5611266a89ff.jpg)
> 
> You can see in the screenshot the NPC info says DisplayID 22380 (Native: 2388). 2388 is his correct modelID. There is no reason to change the modelID.
> 
> So being a rookie to Smart Scripts and the documentation is difficult and lacking crucial details, I don't know if this is a bug or how smart_scripts deals with errors. It happened the most when a specific Watcher moved and then Stitches set data (stitches death) for that Watcher and the Watcher is told to move based on that data being set. I guess I need to know if I should submit a bug report or change the smart scripts.



Haven't been able to look at it again.  I'm not sure if I'm understanding; is the 2nd wave of defenders (after the camp) just bugging out?

LannyE commented:
The 2nd wave bugging out is just one item I noticed not working correctly on your video.  They spawn by the big tree where that forsaken npc was standing (as shown in my screenshot from your video) but they are invisible. They become visible and run to the Rotting Orchard Entrance when Stitches gets past the Night Watch Camp.    

The Darkshire defenders losing is almost impossible without Horde invasion help at the same time.  Your video showed that Watcher Corwin and Watcher Sarys was missing from the darkshire defenders formation so stitches won the fight which makes me think they spawned but didn't move into the formation as they should.  That is why I asked you to look in their spawn locations (Inn and Blacksmith) after Watch Cutford finshes talking to Althea and runs out of Darkshire.   

I also noticed watcher Jan's dead body seemed to have moved after Stitches death which should not be possible since the Evade code specifically checks for him to be alive before moving him back to home position.

I suspect on your installation the SQL to update the Creature_templates for Watchers Corwin, Sarys, Merant, Gelwin, Selkin, and Thayer is not updating them to (SET `AIName` = "SmartAI") didn't work.  At least the behavior looks that way. 

I can't reproduce this so if it is possible would you look to see if Sarys and Corwin are spawning and stand in the location shown in the screenshot where the forsaken guy was standing and let me know what you see when the Town Crier yells that the Night Watch Camp was Overrun?  I would appreciate it. 

@pklloveyou, just for clarity on my end, did your test work fine?

Thanks

Voxstrasza commented:
I tested again, and the 2nd wave just doesn't spawn.  No sarys/corwin.  I assume the Forsaken Commoners are supposed to be hidden as well.

LannyE commented:
Thanks for testing again.  I think I found the/a issue.  I am going to recreate those affected Watchers with new GUIDs and it should hopefully fix those issues.

Voxstrasza commented:
> 
> 
> Thanks for testing again. I think I found the/a issue. I am going to recreate those affected Watchers with new GUIDs and it should hopefully fix those issues.

About Stitches walking speed - I'm 99% sure he's not supposed to be barrelling down the road.  I'm trying to find a video to confirm.

Lastly, the final stand at Darkshire, one of the NPCs is supposed to say this as Stitches is walking up to the town:

```
It'll take more than a rotting corpse to stop us!
```

LannyE commented:
I am almost done with the fixes.  I found a few videos that confirm he is walking slow, basically walking not running.  Will fix that.  

The trash talk from one of the darkshire defenders I can't seem to find a video showing it.  I would love to add it to the correct watcher and have the timing correct and any other trash talk this is missing.  @Voxstrasza, do you know where you saw that and can link that video/forum for me?  I appreciate your helping me to get this as accurate as possible.

Voxstrasza commented:
> 
> 
> I am almost done with the fixes. I found a few videos that confirm he is walking slow, basically walking not running. Will fix that.
> 
> The trash talk from one of the darkshire defenders I can't seem to find a video showing it. I would love to add it to the correct watcher and have the timing correct and any other trash talk this is missing. @Voxstrasza, do you know where you saw that and can link that video/forum for me? I appreciate your helping me to get this as accurate as possible.

https://youtu.be/F4_U_7eMf6g?t=350

Granted, this seems like beta from may 2019 (stitches is even flagged as neutral, to show you how early in development this is).  But I do vaguely remember similar chatter from when I first experienced this event back in 2007.  I'll do some more digging and see if I can find some solid proof.

LannyE commented:
Looking closely at that video it shows Sarys, Corwin, and Althea move into a piece of the V formation they had back then.  The trash talk came from Althea.  This is the first video I have seen that has her involved with the actual defense when Stitches arrives.  Although since it is Classic Beta no one attacks Stitches.  The biggest problem with the Stitches event is that it changed multiple times between Vanilla beta through WOTLK to then be removed and replaced with the lame instanced version in Cata.  

I am trying to fill any gaps from memory of my wow beta and wow vanilla days.  There were times when Stitches was allowed to command the town, with a bunch of NPCs gawking at him the whole time, and then players whining about him curb stomping them in town until he despawned and then I remember the V formation of watchers being added to stop him from commanding the town.  I chose the V formation solution to end the event.  

I like the trash talk from Althea and I don't think she actually aggroes on to Stitches in any situation.  I will add it to her to say (talk within local hearing distance) after the town crier yells to move forward and when the watchers aggro onto Stitches.  I will continue to look for old videos from that era to fine tune the event while I test these changes.   Thanks.

Voxstrasza commented:
@FrancescoBorzi since we have a lack of wotlk specific information about this event, what should we do?

in my opinion, it couldnt hurt to have all the combat NPCs (althea, all the night watch guards, etc) engage stitches in the town and have althea in the V formation leading the defense (it makes sense, she's the highest lvl npc there and she's the leader of the night watch)

LannyE commented:
New commit.  Some of the GUIDs for the event only watchers were already used.  This caused a conflict. 

Other changes:
- Stitches speed reduced to a walk.
- Added new text to Commander Althea to say upon Stitches entering darkshire.
- Changed timing of Stitches yells to reduce the number of yells due to the event lasting longer because Stitches is now walking. 
- Watcher Cutford yells spaced out more due to Stitches walking.

Voxstrasza reviewed (changes requested):
There are some text lines without spaces after the comma, can you add the space after the comma?

Voxstrasza commented:
I'm going to test this.  I will give you my report tomorrow, as we reduced his speed it'll take much longer to test.

LannyE commented:
I understand what you are saying about Althea and it does makes sense.  I pondered over my reasons for leaving Althea out of the formation and here is what I came up with.  

- Videos showing her attack Stitches.  Videos like this one [show](https://www.youtube.com/watch?v=xl8WHdd2iwE&ab_channel=WoWQuests) her standing there while Stitches annihilates everyone.  Unfortunately it is from Wow Classic or it appears so.
- Blizzlike quest giver behavior:  From my experience with Blizzard is they tend to not have quest giver NPCs participate in battle with other NPCs unless they have to, probably due to disrupting the player experience.   Funny though is Watcher Dodds dies in the first picket line.  Although the difference between Althea and Dodds is Althea is a large quest hub.  Her death will interrupt player progression in Darkshire.
-  Military behavior: Commander Althea uses miltitary behavior to interact with her subordinates.  If it were a mob or small rabble of farmers it may make more sense, but her night watcher are disciplined.  She uses Watcher Cutford to put the pickets on basically yellow alert and then relays her orders through the town crier (basically the equivalent to a radio) to the other night watch members. Having a commander take point in the defense of darkshire would put her under duress and thus would not be able to lead or reallocate reserves depending on how the battle is faring.  
- Lore:  Going throught the quest lines show that Darkshire is in a dire situation even before Stitches ever came about.  They are surrounded by an ever-encroaching undead.  I thought about bringing in all Watchers to battle Stitches but due to the situation they are in, over-allocating resources and collapsing your flanks may embolden your enemy to attack from the rear.  Also the player is supposed to feel a duty to protect darkshire based on what they have done.  They should participate.  Without Althea or the players interference the darkshire night watch takes a little over 2 minutes and the loss of 2-3 watchers to down Stitches.  I believe this is a good length to show Stitches was a difficult challenge for the Night Watch.  If Althea is in the formation Stitches will die much quicker, which may make you think that this was not really as dreadful as they made it sound. 
- More blizzlike annoying behavior: Blizzard loves to make us go through tons of peons to get to the commander/leader, ugh.  

Right now I have Watchers: Frazier, Sarys, Corwin, Jan, and Backus in the formation with Jordan and Keefer patrolling into the fray.  Jordan sometimes dies on his own but at any one time there are six watchers attacking Stitches.  Stitches has never prevailed in my tests except when a Horde player runs in and kills the watchers then Stitches commands the town until the event expires.  Stitches attacks the Horde guy too.  

Watchers Jan and Backus are the only two Watchers I cannot find any evidence that they were in the fight against Stitches, but they were the two closest that made some sense to back up the others.  Watcher Mocarski was another I thought of but didn't add.

Anyways let me know what you want done and I will take a look, but I think this is giving the player a very good, if not 100% accurate, blizzlike experience.

Voxstrasza commented:
@LannyE Absolutely beautiful work.  My video of the turning in the quest, and the initial roleplay is still rendering, but here is a video of the first wave of defense against Stitches:  https://youtu.be/LrYBxe_BsCk

First wave checks out.  All good in my book.

Second wave finally spawned.  See video:  https://youtu.be/Pzdjc_036pw

I will post parts 1 and parts 4 of the final battle when the videos are done rendering, and then some notes/what I think could make it better (because we don't have any Blizzlike information to go off of, we should have a little wiggle room here)

Voxstrasza commented:
Here are some of my notes:

The video you linked is not from classic; it's from Vanilla.  It was uploaded May 2019, but you can clearly see it was recorded in the stone age.  The graphics of WoW on that video could pass for a Vanilla private server, but again, it looks like it was recorded pre-2008.  I believe it to be vanilla.

I think we should add her to the V formation and join in on the attack.  There should be a way to disable her quest association for the few minutes she's involved in the event to have this happen; and then once it's done, she either dies and respawns or simply runs back to her original starting point.  There are multiple big NPCs that turn off their quest association to handle an event.

Some of the Watchers in Darkshire didn't despawn, you'll see it in my part 4, final battle video.  Watcher Sarys hangs out in the inn.

Also, do any of the waves of defenders wait for Stitches while in the Ready_1H position and the looping attack/ready animation?  Or do they just wait, with their weapons out and down?

LannyE commented:
Sarys and Corwin despawn when Stitches despawns.  If alive they return to their spawnpoints and wait upon stitches death.  The Town Crier controls the event and once Stitches dies he goes into a 5 minute timer to stop the event snd once that happens it despawns and respawns the correct Watchers.  

I have tried many smart_scripts action_types to get them to do the animation and they don't seem to work.  I will try again.  If you know which Action_Type would make them go into that position please let me know.

Voxstrasza commented:
> 
> 
> Sarys and Corwin despawn when Stitches despawns. If alive they return to their spawnpoints and wait upon stitches death. The Town Crier controls the event and once Stitches dies he goes into a 5 minute timer to stop the event snd once that happens it despawns and respawns the correct Watchers.
> 
> I have tried many smart_scripts action_types to get them to do the animation and they don't seem to work. I will try again. If you know which Action_Type would make them go into that position please let me know.

Basically, once Stitches' body despawns, the NPCs despawn as well?

Would 40_waypoint_reached -> 5_play_emote work?  The emoteID for Ready_1H is either 333 or 45.

LannyE commented:
The event stops and all NPCs that spawned with it (Stitches, Sarys, Corwin, Hutchins, Blomberg, Merant, Thayer, Gelwin, Selkin) all despawn.   The event stops about 5 minutes after Stitches is killed or the event timer (35 minutes) runs out.   On event stop all NPCs that died that were not spawned for the event are respawned.  

I will take a look at the emote.  Thanks for showing me that.

Voxstrasza commented:
> 
> 
> The event stops and all NPCs that spawned with it (Stitches, Sarys, Corwin, Hutchins, Blomberg, Merant, Thayer, Gelwin, Selkin) all despawn. The event stops about 5 minutes after Stitches is killed or the event timer (35 minutes) runs out. On event stop all NPCs that died that were not spawned for the event are respawned.
> 
> I will take a look at the emote. Thanks for showing me that.

Gotcha.  I didn't stick around after the NPCs killed him.  On my next test I will wait 5 minutes.

Is there a cooldown in place for this event to prevent double spawns of Stitches and back-to-back events?

LannyE commented:
35 minutes.  The whole reason having Stitches as a game event is to make it so you cannot have double spawns.  I tested this earlier.  Once the event starts futher turn ins of the Translation to Ello quest will not start it again until the previous stitches event ends.  Basically there is no way for two stitches to be active at the same time.  

I tested it again by turning in the quest four times in a row and the event only ran once with one stitches creature.  

Back to back events is fine because that is what we want.  This is blizzlike.  Waiting for the stitches game event to end (his death and despawn) and then turning in another Translation to Ello quest will start the event all over again.

Voxstrasza commented:
> 
> 
> 35 minutes. The whole reason having Stitches as a game event is to make it so you cannot have double spawns. I tested this earlier. Once the event starts futher turn ins of the Translation to Ello quest will not start it again until the previous stitches event ends. Basically there is no way for two stitches to be active at the same time.
> 
> I tested it again by turning in the quest four times in a row and the event only ran once with one stitches creature.
> 
> Back to back events is fine because that is what we want. This is blizzlike. Waiting for the stitches game event to end (his death and despawn) and then turning in another Translation to Ello quest will start the event all over again.

Great stuff.  Overall amazing work.  Thank you for this!

Voxstrasza commented:
@LannyE any updates ?

LannyE commented:
I added the animations to the Watchers.  I also changed the formation of the Watchers at the Rotting Orchard entrance to be more accurate with Blizzard.  I also had to swap Thayer and Gelwin.  Also I fixed Stitches to run to the road then walk once he turns hostile.  

I was trying to fit Althea in to the defense of darkshire but I just hated the outcome so I started removing watchers to make the fight last longer.  While trying to find anything that would show where she would be in the formation I ran across this [https://wow.gamepedia.com/File:Stitches.jpg](https://wow.gamepedia.com/File:Stitches.jpg) showing the three watchers that attack stitches at the entrance to Darkshire are Frazier, Corwin, And Sarys.  

After seeing that I changed the defenders to be those three.  I increased Sary's and Corwin's damagemodifier to make the fight end there.  I did it to them since they are summoned only for this fight.  

Also Watcher Jordan's pathing is wrong.  He is supposed to be patrolling the road that Stitches uses to come into Darkshire.  He has no waypoint information so I am assuming his pathing is coded into the core and not through smart scripts.   It would be nice to fix his waypoints also.

Voxstrasza commented:
> 
> 
> I added the animations to the Watchers. I also changed the formation of the Watchers at the Rotting Orchard entrance to be more accurate with Blizzard. I also had to swap Thayer and Gelwin. Also I fixed Stitches to run to the road then walk once he turns hostile.
> 
> I was trying to fit Althea in to the defense of darkshire but I just hated the outcome so I started removing watchers to make the fight last longer. While trying to find anything that would show where she would be in the formation I ran across this https://wow.gamepedia.com/File:Stitches.jpg showing the three watchers that attack stitches at the entrance to Darkshire are Frazier, Corwin, And Sarys.
> 
> After seeing that I changed the defenders to be those three. I increased Sary's and Corwin's damagemodifier to make the fight end there. I did it to them since they are summoned only for this fight.
> 
> Also Watcher Jordan's pathing is wrong. He is supposed to be patrolling the road that Stitches uses to come into Darkshire. He has no waypoint information so I am assuming his pathing is coded into the core and not through smart scripts. It would be nice to fix his waypoints also.

Watcher Jordan's pathID is `59410`.  You can find this in the waypoint_data table.

![wpdata](https://i.imgur.com/CCyWMZE.png)

If you aren't already, I suggest downloading Keira3 which makes finding certain things easier.  After selecting a creature, waypoint data can be found in `Spawn Addon`, and it's per GUID.  `Spawn` has a column for MovementType.  An NPC with a waypoint attached to it will have a `2` in MovementType, for WaypointMovement.  Feel free to hit me up on Discord if you have any questions.

I will do another test of this PR, but I think Watcher Jordan can be fixed in a different PR (unless you want to incorporate everything into one, entirely your choice)

Voxstrasza commented:
@LannyE I messaged you on discord!

Voxstrasza commented:
I tested it, everything looks good to me.  I think this is ready for merge, and if you still want to fix Watcher Jordan's waypoints it can be done in a different PR.

Marking to be merged.  Thank you for this @LannyE!

LannyE commented:
I was just about to upload a new commit with Jordan and a minor Fix for Sarys emote returning it to normal after stitches dies.  I can make a new PR for those.

Voxstrasza commented:
> 
> 
> I was just about to upload a new commit with Jordan and a minor Fix for Sarys emote returning it to normal after stitches dies. I can make a new PR for those.

You can push it here and I'll give it one last test before merge, if you want

LannyE commented:
I would love too.  Sorry for all the commits.  It takes about 25 minutes per test so takes a long time to do anything.

Voxstrasza commented:
Tested.  All good.  Will make sure this gets merged before 1 March.
